dependency_support/llvm_register_toolchains_if_not_macos.bzl

```diff
@@ -0,0 +1,30 @@
+# Copyright 2024 The XLS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@llvm_toolchain//:toolchains.bzl", "llvm_register_toolchains")
+
+def _register_toolchain_impl(repository_ctx):
```

Comment:
I wonder if there is a way to declare a system toolchain that point to the macosx runner sysroot similar to what's being done here:
https://github.com/bazel-contrib/toolchains_llvm/blob/3e94f956b6de1c3c6d8c55ccd6f7819925f9770c/tests/WORKSPACE#L90-L95

And then use per-platform toolchain resolution as documented here:
https://github.com/bazel-contrib/toolchains_llvm/tree/3e94f956b6de1c3c6d8c55ccd6f7819925f9770c?tab=readme-ov-file#per-host-architecture-llvm-version

Might be easier than introducing a new rule?

Reply:
It might be useful to have a setting that says to use the platform-provided toolchain (unsupported *nixes also relevant here). How do we feel about [bool_flag](https://github.com/bazelbuild/bazel-skylib/blob/main/rules/common_settings.bzl#L82)s?